src/modules/bruteforce.rs

```diff
@@ -148,3 +157,76 @@ fn approx_color(c: colorgrad::Color) -> yansi::Color {
 
     yansi::Color::Fixed(16 + 36 * r + 6 * g + b)
 }
+
+fn gen_pass_and_hash<T: Rng>(rng: &mut T) -> (String, String) {
+    let pass = format!(
+        "{}{:02}",
+        FirstName(EN).fake::<&str>().to_lowercase(),
+        rng.gen_range(0..99)
+    );
+    let hash = sha256(&pass);
+    (pass, hash)
+}
+
+struct HashGuesser {
+    hash: Vec<u8>,
+    guesses: Vec<u8>,
+    progress: usize,
+    len: usize,
+    rng: ThreadRng,
+}
+
+impl HashGuesser {
+    fn new(hash: &str) -> Self {
+        // the hash string must only contains lowercase hex characters
+        debug_assert!(hash
+            .chars()
+            .all(|c| c.is_ascii_hexdigit() && !c.is_ascii_uppercase()));
```

Comment:
Do we really need this? Let's be real, this will only ever be programmed at exactly this module and I think we have absolutely no other `debug_asserts` :)